src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonInputFormatter.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Formatters.Json;
+
+namespace Microsoft.AspNetCore.Mvc.Formatters
+{
+    /// <summary>
+    /// A <see cref="TextInputFormatter"/> for JSON content that uses <see cref="JsonSerializer"/>.
+    /// </summary>
+    public class SystemTextJsonInputFormatter : TextInputFormatter, IInputFormatterExceptionPolicy
+    {
+        private readonly JsonSerializerOptions _serializerOptions;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="SystemTextJsonInputFormatter"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="MvcOptions"/>.</param>
+        public SystemTextJsonInputFormatter(MvcOptions options)
+        {
+            _serializerOptions = options.SerializerOptions;
```

Comment:
Should this be `JsonSerializerOptions`?

Reply:
I'm slightly jumping the gun here, but we're invariably going to start reading other options from `MvcOptions` to configure the serializer in a couple of releases. Plus it also gets us out of the business of setting up the serializer options since `MvcOptions` initializes it correctly.

Reply:
I was just referrring to the property name. Like why `MvcOptions.SerializerSettings` instead of `MvcOptions.JsonSerializerSettings`.